design/ShaderLanguageFAQ.md

```diff
@@ -0,0 +1,157 @@
+*   It can be faithfully and simply converted to and from SPIR-V
+    “Shader” modules that only use GPU features available in WebGPU.
+
+The W3C WebGPU community group agreed to this direction during the
+February 2020 face-to-face meeting in Redmond.
+
+
+## How do you pronounce “WGSL”?
+
+The W3C WebGPU Community Group does not prescribe a particular
+pronunciation of “WGSL”.
+
+You can say it letter-by-letter, like “W G S L”.
+
+You can say it as a single word, like “wiggzle”, “wiggles”, or “wigsel”.
+
+
+## What does a WGSL shader look like?
+
+WGSL has not yet been specified.
+However, early prototyping work indicates the language can have a look
+and feel as represented in the following examples:
+
+A simple fragment shader:
+
+
+```
+[[location 0]] var<out> gl_FragColor : vec4<f32>;
+
+fn main() -> void {
+    gl_FragColor = vec4<f32>(0.4, 0.4, 0.8, 1.0);
+    return;
+}
+entry_point fragment = main;
+```
+
+
+A code fragment showing if/else:
+
+
+```
+  if (value > 5) {
+    wvar = 14;
+  } elseif (value > 2) {
+    wvar = 12;
+  } else {
+    wvar = 0;
+  }
+```
+
+
+A code fragment showing a switch:
+
+
+```
+switch (a) {
+  case 1: {
+    # The break is implicit
+  }
+  case 2: {
+     fallthrough;
+  }
+  default: {
+  }
+}
+```
+
+
+A code fragment showing a loop:
+
+
+```
+var i : i32 = 0;
+var b : f32 = 2.4;
+loop {
+  break if (i == 4);
+
+  b = b * 3.2;
+  continuing {
+     i = i + 1;
+  }
+}
+```
+
+
+
+## Where can I find the spec?
+
+The WGSL spec will be developed on the WebGPU Community Group area on GitHub.
+See https://github.com/gpuweb/gpuweb/tree/master/wgsl
+
+
+## What browsers support WGSL?
+
+WGSL is the shader language for WebGPU.
+Any browser implementing WebGPU will be required to accept shaders in WGSL.
+Specifically, the WebGPU conformance test suite will use shaders written
+in WGSL.
+
+Browser support for WGSL is expected to be developed over time, along
+with the development of WGSL itself.
+
+
+## Will browsers accept other shader languages for WebGPU?
+
+Accepting a shader language other than WGSL is not part of WebGPU.
+Doing so harms interoperability, as a shader in that other language is
+not portable across browsers.
+
+
+## Where can I find WGSL tooling?
+
+The WebGPU Community Group does not deliver tooling for WGSL.
+However, WebGPU implementers are expected to develop tooling
+for handling WGSL, including parsing, validation, and compilation.
+
+At this time, we expect at least three essentially independent
+implementations of WGSL ingestion as part of browser support for WebGPU.
+Independent implementation serves as an important check on the quality
+of the WGSL and WebGPU specifications.
+
+
+## What about my existing shaders, compilers, and other tooling?
+
+This is also expressed as “the last thing I need is another shader language”.
+(See https://xkcd.com/927/)
+
+For developers invested in the SPIR-V ecosystem for Vulkan and OpenGL,
+the key will be convertibility between WGLSL and SPIR-V.
```

Comment:
"The key?" I don't know what that means.

Rather than being aspirational, please be factual. There will almost certainly eventually be open source compilers from many languages, likely including SPIR-V, to WGSL. The WebGPU standardization group will _not_ produce these compilers.

Reply:
I'm not even sure it's worth calling out SPIR-V explicitly. There will almost certainly be many compilers. They assuredly will intend to be faithful.

Reply:
I read the point here not as "convertibility is possible, and there will be compilers from many languages" but as "simple and clean convertibility with SPIR-V specifically is the goal".

Reply:
I don't think "simple and clean convertibility" is relevant to the audience of this document. "Simple and clean" is a means to an end; the end is portability, both in terms of behavior and performance. There are other ways of achieving that end.

Reply:
The ease of integration with existing shader technologies is *very* much relevant to the audience of this document. You can even see people chiming in about this topic, as a question, *directly* above this comment thread.